python_modules/libraries/dagster-aws/dagster_aws/pipes/clients/ecs.py

```diff
@@ -77,6 +79,10 @@ def run(
             extras (Optional[Dict[str, Any]]): Additional information to pass to the Pipes session in the external process.
             pipes_container_name (Optional[str]): If running more than one container in the task,
                 and using :py:class:`PipesCloudWatchMessageReader`, specify the container name which will be running Pipes.
+            waiter_delay (int): The amount of time in seconds to wait between attempts. Defaults to 6.
```

Comment:
perhaps it will be cleaner to have a single `waiter_config` argument instead? We can type it with a `TypedDict`. Also, I think both fields should be `typing.NotRequired[int]` since we don't always expect the user to set them. 

```python
class WaiterConfig(TypedDict):
    Delay: NotRequired[int]
    MaxAttempts: NotRequired[int]
```